testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py

```diff
@@ -0,0 +1,2860 @@
+        5. Start replication with push pull and continous False.
+        6. Wait until replication done
+        7. Verfiy doc with higher rev id is updated in CBL.
+        8. Now update docs in sync gateway 3 times.
+        9. Start replication with push pull and continous False.
+        10. Wait until replication is done
+        11. As sync-gateway revision id is higher, updates from sync-gateway wins
+    """
+    sg_db = "db"
+    sg_url = params_from_base_test_setup["sg_url"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    sg_mode = params_from_base_test_setup["mode"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    channels = ["replication-channel"]
+    num_of_docs = 10
+
+    # Reset cluster to clean the data
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # Create bulk doc json
+    if attachments:
+        db.create_bulk_docs(num_of_docs, "replication", db=cbl_db, channels=channels, attachments_generator=attachment.generate_2_png_10_10)
+    else:
+        db.create_bulk_docs(num_of_docs, "replication", db=cbl_db, channels=channels)
+    sg_client = MobileRestClient()
+
+    # Start and stop continuous replication
+    replicator = Replication(base_url)
+    sg_client.create_user(sg_admin_url, sg_db, name="autotest", password="password", channels=channels)
+    session, replicator_authenticator, repl = replicator.create_session_configure_replicate(
+        baseUrl=base_url, sg_admin_url=sg_admin_url, sg_db=sg_db, channels=channels, sg_client=sg_client, cbl_db=cbl_db, sg_blip_url=sg_blip_url, username="autotest", password="password", replication_type="push_pull", continuous=False)
+    sg_docs = sg_client.get_all_docs(url=sg_url, db=sg_db, auth=session)
+    sg_docs = sg_docs["rows"]
+
+    if highrev_source == 'cbl':
+        sg_client.update_docs(url=sg_url, db=sg_db, docs=sg_docs, number_updates=1, auth=session)
+        db.update_bulk_docs(cbl_db, number_of_updates=2)
+    else:
+        sg_client.update_docs(url=sg_url, db=sg_db, docs=sg_docs, auth=session, number_updates=2)
+        db.update_bulk_docs(cbl_db)
+
+    replicator.configure_and_replicate(source_db=cbl_db, replicator_authenticator=replicator_authenticator, target_url=sg_blip_url, continuous=False,
+                                       channels=channels)
+    if sg_mode == "di":
+        replicator.configure_and_replicate(source_db=cbl_db, replicator_authenticator=replicator_authenticator, target_url=sg_blip_url, continuous=False,
+                                           channels=channels)
+    cbl_doc_ids = db.getDocIds(cbl_db)
+    cbl_docs = db.getDocuments(cbl_db, cbl_doc_ids)
+
+    sg_docs
```

Comment:
In keywords/SyncGateway.py there is a restart_sync_gateways method. Why do we have the same thing here?

Reply:
The reason is that I need to restart first sync gateway of the list . I had to use cluster sync_gateway restart which calls libraries/testkit/syncgateway.py